components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -934,37 +965,34 @@ def load_execution_container_name(self):
             # Accept configured value for debug purposes
             return
 
-        os_release = dotenv_values(self._os_release_file_path)
-        if "ubuntu" == os_release["ID"]:
-            self.execution_container = (
-                f"clp-execution-x86-{os_release['ID']}-{os_release['VERSION_CODENAME']}:main"
-            )
-        else:
-            raise NotImplementedError(
-                f"Unsupported OS {os_release['ID']} in {OS_RELEASE_FILE_PATH}"
-            )
+        if self._image_id_path.exists():
+            with open(self._image_id_path) as image_id_file:
+                self.execution_container = image_id_file.read().strip()
 
-        self.execution_container = "ghcr.io/y-scope/clp/" + self.execution_container
+        if not bool(self.execution_container):
+            with open(self._version_file_path) as version_file:
+                package_version = version_file.read().strip()
+            self.execution_container = f"ghcr.io/y-scope/clp/clp-package:{package_version}"
 
     def get_shared_config_file_path(self) -> pathlib.Path:
         return self.logs_directory / CLP_SHARED_CONFIG_FILENAME
 
-    def get_runnable_components(self) -> Set[str]:
+    def get_deployment_type(self) -> DeploymentType:
```

Comment:
i feel i should alphabetize the methods though the file would be pretty messed up. it might be better to reorder after we're done with the review, or in a separate PR